ebos/eclproblem.hh

```diff
@@ -3018,20 +2914,35 @@ private:
 
     EclActionHandler actionHandler_;
 
-    std::vector<bool> freebcX_;
-    std::vector<bool> freebcXMinus_;
-    std::vector<bool> freebcY_;
-    std::vector<bool> freebcYMinus_;
-    std::vector<bool> freebcZ_;
-    std::vector<bool> freebcZMinus_;
-
-    bool nonTrivialBoundaryConditions_;
-    std::vector<RateVector> massratebcX_;
-    std::vector<RateVector> massratebcXMinus_;
-    std::vector<RateVector> massratebcY_;
-    std::vector<RateVector> massratebcYMinus_;
-    std::vector<RateVector> massratebcZ_;
-    std::vector<RateVector> massratebcZMinus_;
+    template<class T>
+    struct BCData
+    {
+        std::array<std::vector<T>,6> data;
+
+        void resize(size_t size, T defVal)
+        {
+            for (auto& d : data)
+                d.resize(size, defVal);
+        }
+
+        const std::vector<T>& operator()(FaceDir::DirEnum dir) const
+        {
+            if (dir == FaceDir::DirEnum::Unknown)
+                throw std::runtime_error("Tried to acess BC data for the 'Unknown' direction");
+            int idx = 31 - __builtin_clz(static_cast<int>(dir));
```

Comment:
I **really** don't think we should use compiler intrinsics.  I'd prefer we use something more portable&mdash;e.g., our home-grown [`count_leading_zeros()`](https://github.com/OPM/opm-common/blob/f414cdd30e5792b1198cfe08082a7605b28668b8/src/opm/input/eclipse/Schedule/WriteRestartFileEvents.cpp#L61) function or its 32-bit equivalent.

Reply:
if that's a concern i'd rather just use the 2 line while loop. okay?